atest/robot/standard_libraries/builtin/repeat_keyword.robot

```diff
@@ -73,6 +81,18 @@ Check Repeated Messages
     Run Keyword If    ${count} == 0    Check Log Message    ${kw.msgs[0]}    Keyword 'This is not executed' repeated zero times.
     Run Keyword If    ${count} != 0    Should Be Equal As Integers    ${kw.msg_count}    ${count}
 
+Check Repeated Messages With Time
+    [Arguments]    ${kw}    ${expect_count}    ${msg}=${None}
+    # Cannot test exactly how many times kw is run because it depends on
+    # interpreter speed.
+    Should Be True    ${kw.kw_count} <= ${expect_count}
+    : FOR    ${i}    IN RANGE    ${kw.kw_count}
+    \    Check Log Message    ${kw.msgs[${i}]}
+    \    ...    Repeating keyword, round ${i+1}, remaining*    pattern=yes
+    \    Check Log Message    ${kw.kws[${i}].kws[0].msgs[0]}    ${msg}
+    Run Keyword If    ${kw.kw_count} == 0    Check Log Message    ${kw.msgs[0]}    Keyword 'This is not executed' repeated zero times.
+    Run Keyword If    ${kw.kw_count} != 0    Should Be Equal As Integers    ${kw.msg_count}    ${kw.kw_count}
```

Comment:
This is pretty complicated. Wouldn't it be enough to test that the keyword has been executed at least once? Trying to guess how many times it is actually run is hard and creates complex and/or fragile tests.